test/src/problems/BasicExternalProblem.C

```diff
@@ -33,6 +33,23 @@ BasicExternalProblem::addExternalVariables()
   _heat_source_var = _aux->getFieldVariable<Real>(0, "heat_source").number();
 }
 
+void
+BasicExternalProblem::initializeVariable()
+{
+  auto & solution = _aux->solution();
+  const auto sys_number = _aux->number();
+  const auto & mesh = _mesh.getMesh();
+
+  for (const auto elem_ptr : mesh.element_ptr_range())
+  {
+    auto dof_idx = elem_ptr->dof_number(sys_number, _heat_source_var, 0);
+    solution.set(dof_idx, 12345);
```

Comment:
?

Reply:
I would use the ExternalProblem directly rather than those test objects to test the fix

Reply:
I'm not sure this is the fix I had in mind -- we don't necessarily have an initial value to put into this external variable until we run our OpenMC simulation. The problem is that even after we run OpenMC, the user objects are lagged by one time step (with the current MOOSE, not including any changes in this PR). This fix seems to be just manually adding an insertion point to initialize the variable (which for us would mean an extra OpenMC run -- as opposed to MOOSE just realizing we already ran OpenMC and properly keeping things up to date).

Is my understanding correct?

Reply:
I believe you're correct. When I looked at the two input files you included it looked like for the external case heat_source was never initialized and so it defaulted to 0. Then in the first iteration layered_integral was calculated before heat_source was ever assigned a value which is why it was off for that first timestep. That's why I assumed it was an issue with initialization.

Reply:
So I did some more digging and I think I may have changed the execute_on parameter during some of my debugging but forgot to change it back. So I went back and ran the two original input files you included and when both the aux kernel and user object are set to execute on timestep_end I'm getting the same results for both. The only difference being that for the external user object heat_source is 0 for t = 0 unless I initialize heat_source. I'm wondering if this issue may have been resolved in a different PR and adding an option to initialize and external variable was the only thing left to fix? I was going to try and do some testing with a time dependent problem and see if it still works.

Reply:
@aprilnovak So I ran similar problems that were time-dependent and both the external and non-external user object versions gave the same results. The only difference was that for the external user object, heat_source is initialized to 0. I think this issue may have been resolved earlier. Unfortunately I'm not sure which PR may have fixed it. For reference here are the two examples I ran. For the external one I modified the SyncSolutions function in BasicExternalProblem.C so that it set heat_source to 12345*time(). If you run into any similar issues let me know and I'll take another look!
[time-depenent example.zip](https://github.com/user-attachments/files/18170973/time-depenent.example.zip)